exporters/metric/cortex/auth.go

```diff
@@ -88,3 +92,99 @@ func (e *Exporter) addBearerTokenAuth(req *http.Request) error {
 
 	return nil
 }
+
+// buildClient returns a http client that uses TLS and has the user-specified proxy and
+// timeout.
+func (e *Exporter) buildClient() (*http.Client, error) {
+	// Create a TLS Config struct for use in a custom HTTP Transport.
+	tlsConfig, err := e.buildTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+
+	// Convert proxy url to proxy function for use in the created Transport.
+	if e.config.ProxyURL != "" {
+		proxyURL, err := url.Parse(e.config.ProxyURL)
+		if err != nil {
+			return nil, err
+		}
+		proxy := http.ProxyURL(proxyURL)
+		transport.Proxy = proxy
+	}
+
+	client := http.Client{
+		Transport: transport,
+		Timeout:   e.config.RemoteTimeout,
+	}
+	return &client, nil
+}
+
+// buildTLSConfig creates a new TLS Config struct with the properties from the exporter's
+// Config struct.
+func (e *Exporter) buildTLSConfig() (*tls.Config, error) {
+	tlsConfig := &tls.Config{}
+	if e.config.TLSConfig == nil {
+		return tlsConfig, nil
+	}
+
+	// Set the server name if it exists.
+	if e.config.TLSConfig["server_name"] != "" {
+		tlsConfig.ServerName = e.config.TLSConfig["server_name"]
+	}
+
+	// Set InsecureSkipVerify. Viper reads the bool as a string since it is in a map.
+	parsedBool, err := strconv.ParseBool(e.config.TLSConfig["insecure_skip_verify"])
+	if err != nil {
+		return nil, err
+	}
+	tlsConfig.InsecureSkipVerify = parsedBool
```

Comment:
`"insecure_skip_verify"` may not exist in the `TLSConfig` making this parse error unnecessarily. The map should be checked if it contains the key prior to trying to parse it. I.e.

```suggestion
	if isv, ok := e.config.TLSConfig["insecure_skip_verify"]; ok {
		var err error
		if tlsConfig.InsecureSkipVerify, err = strconv.ParseBool(isv); err != nil {
			return nil, err
		}
	}
```